### PR TITLE
Updates Examples With New Package Names, Adds Warning, Fix Thread Panic

### PR DESCRIPTION
**Pull request by dvonthenen**
## Proposed changes

Addresses: https://github.com/deepgram/deepgram-go-sdk/issues/251

This PR does three things:
- Implements a fix for https://github.com/deepgram/deepgram-go-sdk/issues/251. In a situation when we panic in a thread we created, we emit the error and close the client immediately. This will cause the client to reconnect properly in this situation.
- Implements the Warning messages and callback.
- Finishes updating all examples to migrate from the deprecated package names to the new ones. This is why you see deletion of the `//lint:ignore` comment. We are no longer using deprecated package names.

Tested on all examples.

## Types of changes

What types of changes does your code introduce to the community Go SDK?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update or tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I have lint'ed all of my code using repo standards
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

## Further comments

NA

## Summary by CodeRabbit

- **New Features**
  - Introduced support for the `listen/v1/rest` and `listen/v1/websocket` API packages.
  - Added new error handling and recovery mechanisms for WebSocket connections.

- **Improvements**
  - Enhanced clarity and consistency in log messages and method names across the `speak` and `listen` clients.
  - Updated SDK version to `v1.4.0`.

- **Deprecations**
  - Deprecated certain client aliases and functions in favor of more descriptive names.

- **Refactor**
  - Unified import paths and updated client instantiation methods for better readability and efficiency.

### CODE REVIEW

File: examples/speech-to-text/websocket/microphone/main.go [hunk 1]
```diff
@@ -150,7 +150,7 @@ func main() {
 	}
 
 	// create a Deepgram client
-	dgClient, err := client.New(ctx, "", cOptions, tOptions, callback)
+	dgClient, err := client.NewWebSocket(ctx, "", cOptions, tOptions, callback)
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply examples/speech-to-text/websocket/microphone/main.go:153]: This already has trace statements:
```
	if err != nil {
		fmt.Println("ERROR creating LiveTranscription connection:", err)
		return
	}
```

File: examples/text-to-speech/rest/stream/woodchuck/main.go [hunk 2]
```diff
@@ -12,7 +12,7 @@ import (
 
 	prettyjson "github.com/hokaccha/go-prettyjson"
 
-	speak "github.com/deepgram/deepgram-go-sdk/pkg/api/speak/v1" //lint:ignore
+	api "github.com/deepgram/deepgram-go-sdk/pkg/api/speak/v1/rest"
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply examples/text-to-speech/rest/stream/woodchuck/main.go:15]: this was updated to match the naming convention. good catch!

File: examples/manage/usage/main.go [hunk 3]
```diff
@@ -14,7 +14,7 @@ import (
 
 	api "github.com/deepgram/deepgram-go-sdk/pkg/api/manage/v1"
 	interfaces "github.com/deepgram/deepgram-go-sdk/pkg/api/manage/v1/interfaces"
-	client "github.com/deepgram/deepgram-go-sdk/pkg/client/rest"
+	client "github.com/deepgram/deepgram-go-sdk/pkg/client/manage"
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply examples/manage/usage/main.go:17]: The `pkg/client/rest/v1` isnt deprecated. It's just the direct use of the helper file `pkg/client/rest/client.go`.

File: examples/manage/scopes/main.go [hunk 4]
```diff
@@ -11,7 +11,7 @@ import (
 
 	api "github.com/deepgram/deepgram-go-sdk/pkg/api/manage/v1"
 	interfaces "github.com/deepgram/deepgram-go-sdk/pkg/api/manage/v1/interfaces"
-	client "github.com/deepgram/deepgram-go-sdk/pkg/client/rest"
+	client "github.com/deepgram/deepgram-go-sdk/pkg/client/manage"
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply examples/manage/scopes/main.go:14]: The pkg/client/rest/v1 isnt deprecated. It's just the direct use of the helper file pkg/client/rest/client.go.

File: examples/manage/usage/main.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (dvonthenen) [reply examples/manage/usage/main.go:17]: does that make sense? We dont want users to create a rest client directly, which is why we introduced the `manage` client. we want a level of indirection on the client.

File: pkg/client/speak/client.go [hunk 5]
```diff
@@ -55,6 +57,16 @@ func New(apiKey string, options *interfaces.ClientOptions) *speakv1rest.Client {
...
 }
 
+/*
+NewRESTWithDefaults creates a new speak client with all default options
+
+Notes:
+  - The Deepgram API KEY is read from the environment variable DEEPGRAM_API_KEY
+*/
+func NewRESTWithDefaults() *speakv1rest.Client {
+	return speakv1rest.NewWithDefaults()
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply pkg/client/speak/client.go:67]: The `analyze` and `manage` clients are only REST and therefore don't need to to have a `NewRESTWithDefaults`

**Review by jpvajda — APPROVED** [review]

**Pull request merged**